x-pack/plugins/security_solution/public/management/services/exceptions_list/exceptions_list_api_client.ts

```diff
@@ -0,0 +1,204 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import {
+  CreateExceptionListItemSchema,
+  CreateExceptionListSchema,
+  ExceptionListItemSchema,
+  ExceptionListSummarySchema,
+  FoundExceptionListItemSchema,
+  ListId,
+  UpdateExceptionListItemSchema,
+} from '@kbn/securitysolution-io-ts-list-types';
+import { EXCEPTION_LIST_ITEM_URL, EXCEPTION_LIST_URL } from '@kbn/securitysolution-list-constants';
+import { HttpStart } from 'kibana/public';
+import { MANAGEMENT_DEFAULT_PAGE, MANAGEMENT_DEFAULT_PAGE_SIZE } from '../../common/constants';
+
+/**
+ * A generic class to be used for each artifact type.
+ * It follow the Singleton pattern.
+ * Please, use the getInstance method instead of creating a new instance when using this implementation.
+ */
+export class ExceptionsListApiClient {
+  private static instance: Map<ListId, ExceptionsListApiClient>;
```

Comment:
@paul-tavares Noticed this `ListId` is too generic (string). Do you know if we already have a management supported list of exceptions lists type somewhere? If not, I will create it to ensure only instances with allowed lists are created using this class

Reply:
yeah, we have this one:

https://github.com/elastic/kibana/blob/997988fac202d130a5b519fad4249efbdff378f1/x-pack/plugins/security_solution/common/endpoint/service/artifacts/constants.ts#L18-L22


But: maybe being generic `string` is desirable. I know this class is for our use right now, but others in security solution could also take advantage of it to manage any list via the API - and if that ever happens, it could be promoted to `security_solution/public/comon/*`

Reply:
yeah, that make sense!